lib/node_modules/@stdlib/random/base/mt19937/lib/rand_uint32.js

```diff
@@ -20,33 +20,33 @@
 
 // MODULES //
 
-var INT32_MAX = require( '@stdlib/constants/math/int32-max' );
+var UINT32_MAX = require( '@stdlib/constants/math/uint32-max' );
 var floor = require( '@stdlib/math/base/special/floor' );
 
 
 // VARIABLES //
 
-var MAX = INT32_MAX - 1;
+var MAX = UINT32_MAX - 1;
 
 
 // MAIN //
 
 /**
-* Returns a pseudorandom integer on the interval \\([1, 2^{31}-1)\\).
+* Returns a pseudorandom integer on the interval \\([1, 2^{32})\\).
```

Comment:
`2^{32} - 1`. You subtract one at L29 and then you add one back in L45. As `MAX*Math.random()` cannot exceed `UINT32_MAX - 1`, then adding one cannot exceed `UINT32_MAX`, which is equal to `2^{32} - 1`.